pyproject.toml

```diff
@@ -63,10 +63,10 @@ dev = [
   "wheel==0.38.4",
 ]
 docs = [
-  "Sphinx==4.5.0",
-  "myst-parser==0.18.1",
-  "sphinxcontrib-serializinghtml==1.1.5",
-  "sphinx-rtd-theme==1.1.1",
+  "Sphinx==7.2.6",
+  "myst-parser==2.0.0",
+  "sphinxcontrib-serializinghtml==1.1.10",
+  "sphinx-rtd-theme==2.0.0",
```

Comment:
You're a brave man. I attempted all kinds of version mixes and couldn't get `make html` in the docs directory to work without spewing errors. Though, after I messed around some, I suspect they were probably just warnings (which persist prior to this PR). Have you confirmed it builds?

Reply:
Builds locally. I see we don't have branch builds active in RTD, seeing how to turn that on.